File: solutions/csharp/prime-factors/1/PrimeFactors.cs
public static class PrimeFactors
{
    public static long[] Factors(long number)
    {
        List<long> primes = new List<long>();
        for (int possiblePrime = 2; number > 1; possiblePrime++)
        {
            for (; number % possiblePrime == 0; number /= possiblePrime)
                primes.Add(possiblePrime);
        }
        return primes.ToArray();
    }

}

File: solutions/cpp/prime-factors/1/prime_factors.cpp
#include "prime_factors.h"
namespace prime_factors {
std::vector<long long> of(long long num) {
  std::vector<long long> primes{};
  for (long candidate = 2; num > 1; candidate++) {
    for (; num % candidate == 0; num /= candidate) {
      primes.emplace_back(candidate);
    }
  }
  return primes;
}
} // namespace prime_factors
